api/envoy/extensions/filters/network/http_connection_manager/v4alpha/http_connection_manager.proto

```diff
@@ -506,6 +507,44 @@ message HttpConnectionManager {
   //
   // 3. Tracing decision (sampled, forced, etc) is set in 14th byte of the UUID.
   RequestIDExtension request_id_extension = 36;
+
+  // Configuration of local reply returned by Envoy. Allows to specify mappings and format of
+  // response.
+  LocalReplyConfig local_reply_config = 38;
+}
+
+message LocalReplyConfig {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.extensions.filters.network.http_connection_manager.v3.LocalReplyConfig";
+
+  // Configuration of list of mappers which allows to filter and change local response.
+  // The client will iterate through mappers until first match.
+  repeated ResponseMapper mapper = 1;
+
+  // Allows to define custom format of local reply.
+  // It is allowed to use :ref:`command operators <config_access_log_command_operators>`
+  // which will be resolved to actual data.
+  type.v3.StringOrJson format = 2;
+}
+
+message ResponseMapper {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.extensions.filters.network.http_connection_manager.v3.ResponseMapper";
+
+  // Filter is used to determine if the response should be rewritten.
+  config.accesslog.v3.AccessLogFilter filter = 1 [(validate.rules).message = {required: true}];
+
+  // Rewriter defines how to rewrite determined response.
+  ResponseRewriter rewriter = 2 [(validate.rules).message = {required: true}];
+}
+
+// Configuration of new value for matched local response.
+message ResponseRewriter {
+  option (udpa.annotations.versioning).previous_message_type =
+      "envoy.extensions.filters.network.http_connection_manager.v3.ResponseRewriter";
+
+  // Status code for matched response.
```

Comment:
Should this be in a `oneof`?

Reply:
ResponseRewriter could re-write status_code, text_body and headers.